diagnostic.md

```diff
@@ -2,37 +2,51 @@
 
 What are three advantages of using Ember?
 
-```sh
-# your answer here
+```md
+A routing system that allows for a single page application allowing for
+  back functionality and boomarking
+Less AJAX
+More use of the cache because of less AJAX, yielding more responsiveness
 ```
 
 What is the templating system used in Ember? How it is related to
 handlebars?
 
-```sh
-# your answer here
+```md
+Each template is written in Handlebars. The template is the HTML content that
+corresponds to the route.
 ```
 
 Name at least two layers of an Ember application (1.0 or 2.0):
 
-```sh
-# your answer here
+```md
+Routes
+Templates
+Models
 ```
 
 How does a component differ from an Ember template?
 
-```sh
-# your answer here
+```md
+Still kind of confused about this. Components come from the template and don't
+have full access to the route. They are smaller pieces of code and thus, more
+modular.
```

Comment:
Yes. They also contain logic/data that is linked to the component template.
1/1